src/rabbit_log.erl

```diff
@@ -73,14 +73,13 @@ log(Category, Level, Fmt, Args) when is_list(Args) ->
     end,
     lager:log(Sink, Level, self(), Fmt, Args).
 
-make_internal_sink_name(rabbit_log_connection) -> rabbit_log_connection_lager_event;
-make_internal_sink_name(rabbit_log_channel) -> rabbit_log_channel_lager_event;
-make_internal_sink_name(rabbit_log_mirroring) -> rabbit_log_mirroring_lager_event;
-make_internal_sink_name(rabbit_log_queue) -> rabbit_log_queue_lager_event;
-make_internal_sink_name(rabbit_log_federation) -> rabbit_log_federation_lager_event;
-make_internal_sink_name(rabbit_log_upgrade) -> rabbit_log_upgrade_lager_event;
-make_internal_sink_name(Category) ->
-    lager_util:make_internal_sink_name(Category).
+make_internal_sink_name(connection) -> rabbit_log_connection_lager_event;
+make_internal_sink_name(channel)    -> rabbit_log_channel_lager_event;
+make_internal_sink_name(mirroring)  -> rabbit_log_mirroring_lager_event;
+make_internal_sink_name(queue)      -> rabbit_log_queue_lager_event;
+make_internal_sink_name(federation) -> rabbit_log_federation_lager_event;
+make_internal_sink_name(upgrade)    -> rabbit_log_upgrade_lager_event;
+make_internal_sink_name(Category)   -> error({unknown_category, Category}).
```

Comment:
The error here is to force people to use the defined set of categories. So running a test will show that the code has an error. Otherwise we would have lost logs and that's not what people generally want. If someone wants to have a custom sink, should it go through `rabbit_log:log`? I think they better use lager parse-transforms instead.
This function here is left for backwards compatibility and ideally should be deleted, for my opinion.